src/rgw/rgw_sts.h

```diff
@@ -0,0 +1,49 @@
+#ifndef CEPH_RGW_STS_H
+#define CEPH_RGW_STS_H
+
+class RGWSTS
+{
+  static const string key;
+  static constexpr int policy_plain_text_size = 2048;
+
+  int duration;
+  int policy_size;
+  std::string role_id;
+  std::string policy;
+  std::string access_key_id;
+  std::string secret_access_key;
+  std::string session_token;
+  std::map<std::string, std::string> saml_keys;
+  struct timeval expiration;
+
+public:
+  RGWSTS (int duration, std::string role_id, std::string policy, std::map<std::string, std::string> saml_keys) :
+    duration(std::move(duration)),
+    role_id(std::move(role_id)),
+    policy(std::move(policy)),
+    saml_keys(std::move(saml_keys)) {}
+
+    void build_output();
+
+    std::string get_access_key_id() {
```

Comment:
I don't mind the get methods per se, (but if you are going to use them make them return const references rather than copying the strings.)

It might be better to just have this be a function that takes some parameters and returns a struct containing just the STS output.